From 6211fdd4f840e6b95074a4a737044ca7fc581904 Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Thu, 01 Oct 2020 15:23:14 +0300
Subject: [PATCH] [SW-20246] preboot: add boot device status

Add Boot device status register set/clr API in preboot,
and indicate that device status register is used by setting
enable bit in the register.

Change-Id: If772cbbb9c2066193e558f17ca08e7c4f6aa8720
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/common/boot_if.c b/common/boot_if.c
index 6d848ace..e0f9e21 100644
--- a/common/boot_if.c
+++ b/common/boot_if.c
@@ -35,15 +35,26 @@
 	regmap_writel(val, mmPSOC_GLOBAL_CONF_CPU_BOOT_STATUS);
 }
 
-/* We support only ERR0 for now */
-void cpu_boot_err_clrset(u32 mask, bool set)
+static void cpu_boot_reg_clrset(u32 reg, u32 mask, bool set)
 {
-	u32 val = regmap_readl(mmCPU_BOOT_ERR0) & ~mask;
+	u32 val = regmap_readl(reg) & ~mask;
 
 	if (set)
 		val |= mask;
 
-	regmap_writel(val, mmCPU_BOOT_ERR0);
+	regmap_writel(val, reg);
+}
+
+/* We support only ERR0 for now */
+void cpu_boot_err_clrset(u32 mask, bool set)
+{
+	cpu_boot_reg_clrset(mmCPU_BOOT_ERR0, mask, set);
+}
+
+/* We support only STS0 for now */
+void cpu_boot_dev_sts_clrset(u32 mask, bool set)
+{
+	cpu_boot_reg_clrset(mmCPU_BOOT_DEV_STS0, mask, set);
 }
 
 void cpu_boot_rsp_set(enum cpu_msg_status sts)
diff --git a/include/boot_if.h b/include/boot_if.h
index e0f54c0..52bb58b 100644
--- a/include/boot_if.h
+++ b/include/boot_if.h
@@ -13,6 +13,7 @@
 void cpu_boot_cmd_clr_before_wfe(void);
 void cpu_boot_sts_set(u32 val);
 void cpu_boot_err_clrset(u32 mask, bool set);
+void cpu_boot_dev_sts_clrset(u32 mask, bool set);
 void cpu_boot_rsp_set(enum cpu_msg_status sts);
 
 #endif /* _HL_BOOT_IF_H_ */
diff --git a/preboot/preboot.c b/preboot/preboot.c
index 5882bfa..5f9188a 100644
--- a/preboot/preboot.c
+++ b/preboot/preboot.c
@@ -193,6 +193,8 @@
 	cpu_boot_sts_set(CPU_BOOT_STATUS_IN_PREBOOT);
 	/* Enable the errors communication */
 	cpu_boot_err_clrset(CPU_BOOT_ERR0_ENABLED, true);
+	/* Enable the device status communication */
+	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_ENABLED, true);
 
 	/*
 	 * Note: we assume CPU frequency is not changed while CPU is running,
